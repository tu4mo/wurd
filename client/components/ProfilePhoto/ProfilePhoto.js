import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import './ProfilePhoto.scss'

const ProfilePhoto = ({ isMe, size, url, username }) =>
  <div
    className={classnames('profile-photo', {
      [`profile-photo--${size}`]: size
    })}
  >
    <img
      alt={username}
      className="profile-photo__image"
      src={url}
      title={username}
    />
    {isMe &&
      <a
        className="profile-photo__change"
        href="https://www.gravatar.com"
        target="_blank"
      >
        Change Gravatar
      </a>}
  </div>

ProfilePhoto.propTypes = {
  isMe: PropTypes.bool,
  size: PropTypes.oneOf(['small', 'tiny']),
  url: PropTypes.string,
  username: PropTypes.string
}

export default ProfilePhoto
