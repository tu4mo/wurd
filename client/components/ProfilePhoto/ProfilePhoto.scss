.profile-photo {
  border-radius: 50%;
  display: inline-block;
  height: 160px;
  overflow: hidden;
  position: relative;
  vertical-align: middle;
  width: 160px;

  &:hover {
    .profile-photo__change {
      visibility: visible;
    }
  }

  &--small {
    height: 40px;
    width: 40px;
  }

  &__change {
    align-items: center;
    background-color: rgba(255, 255, 255, 0.8);
    bottom: 0;
    display: flex;
    justify-content: center;
    left: 0;
    line-height: 1;
    position: absolute;
    right: 0;
    text-align: center;
    text-decoration: none;
    top: 0;
    visibility: hidden;
  }

  &__image {
    display: block;
    height: 100%;
    width: 100%;
  }
}
