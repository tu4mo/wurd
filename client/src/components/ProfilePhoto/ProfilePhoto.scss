.profile-photo {
  border-radius: 50%;
  display: block;
  height: 160px;
  width: 160px;

  &--small {
    height: 40px;
    width: 40px;
  }
}
