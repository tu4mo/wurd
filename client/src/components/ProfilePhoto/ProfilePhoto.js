import React from 'react'
import PropTypes from 'prop-types'
import profile from '../../assets/profile.png'
import './ProfilePhoto.scss'

const ProfilePhoto = ({ onUpload, size, url, username }) => {
  const classNames = ['profile-photo']

  if (size === 'small') {
    classNames.push('profile-photo--small')
  }

  return (
    <div className={classNames.join(' ')}>
      <img
        alt={username}
        className="profile-photo__image"
        src={url}
        title={username}
      />
      {onUpload &&
        <div className="profile-photo__upload">
          Upload Photo
          <input
            accept=".gif,.jpg,.png"
            className="profile-photo__file-input"
            onChange={onUpload}
            type="file"
          />
        </div>}
    </div>
  )
}

ProfilePhoto.propTypes = {
  onUpload: PropTypes.func,
  size: PropTypes.oneOf(['small']),
  url: PropTypes.string,
  username: PropTypes.string
}

ProfilePhoto.defaultProps = {
  url: profile
}

export default ProfilePhoto
