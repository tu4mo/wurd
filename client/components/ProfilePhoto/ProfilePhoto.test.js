import React from 'react'
import renderer from 'react-test-renderer'

import ProfilePhoto from './ProfilePhoto'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <ProfilePhoto
        className="test"
        isMe
        size="small"
        url="https://www.test.com"
        username="test"
      />
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
