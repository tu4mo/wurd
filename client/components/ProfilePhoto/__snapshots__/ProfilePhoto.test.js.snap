// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="test sc-bdVaJa gUwqsU"
>
  <img
    alt="test"
    className="image"
    src="https://www.test.com"
    title="test"
  />
  <a
    className="change"
    href="https://www.gravatar.com"
    target="_blank"
  >
    Change Gravatar
  </a>
</div>
`;
